### PR TITLE
[Auto lint] Refactor sdk.python.kfp.dsl by yapf and docformatter

### PR DESCRIPTION
**Pull request by numerology**
This is the auto-lint change made by yapf with configuration in `.style.yapf` and running `docformatter --in-place --wrap-summaries 80 --wrap-descriptions 80 --blank dsl/*.py` to lint the docstring. Please see #2442 for some background.

It seems like docformatter is not very aggressive in wrapping long lines to comply with length limit, but I think overall the readability is preserved well. Maybe some manual adjustment could make it looks better.

Please let me know how does it look. @neuromage @IronPan @hongye-sun @gaoning777 @gaoning777 @SinaChavoshi @elikatsis

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/2498)

### CODE REVIEW

File: sdk/python/kfp/dsl/types.py [hunk 1]
```diff
@@ -17,187 +17,215 @@
...
+    return False
+  type_name = checked_type_name
+  for type_property in checked_type[type_name]:
+    if type_property not in expected_type[type_name]:
+      print(
+          type_name + ' has a property ' + str(type_property) +
+          ' that the latter does not.'
+      )
+      return False
+    if checked_type[type_name][type_property] != expected_type[type_name
```
Comment (Ark-kun) [inline sdk/python/kfp/dsl/types.py]: I think that this line looks a little strange.

File: sdk/python/kfp/dsl/types.py [hunk 2]
```diff
@@ -17,187 +17,215 @@
...
-	return True
+  if not isinstance(payload, dict) or len(payload) != 1:
+    return False
+  for type_name in payload:
+    if not isinstance(payload[type_name], dict):
+      return False
+    property_types = (int, str, float, bool)
+    property_value_types = (int, str, float, bool, dict)
+    for property_name in payload[type_name]:
+      if not isinstance(property_name, property_types) or not isinstance(
```
Comment (Ark-kun) [inline sdk/python/kfp/dsl/types.py]: Maybe we should wrap before `or` here.

**Comment by Ark-kun** [comment]
The public official Google style guide seems to have slightly different style: http://google.github.io/styleguide/pyguide.html#s3.4-indentation https://github.com/google/styleguide/blob/gh-pages/pyguide.md#s3.4-indentation 

I also found Google's official `yapf` configuration: https://github.com/google/yapf/blob/126fc7eb85dec8fe34104ff15ed64c3a23c9534a/yapf/yapflib/style.py#L398 which can be used by specifying `--style google`.

Let's try it.

File: sdk/python/kfp/dsl/types.py (hunk 1 above)
Comment (numerology) [reply sdk/python/kfp/dsl/types.py]: Agree. Auto-format is not perfect.

File: sdk/python/kfp/dsl/types.py (hunk 2 above)
Comment (numerology) [reply sdk/python/kfp/dsl/types.py]: I guess the reason was that it tries to fix 80 chars line limit.

**Comment by numerology** [comment]
> The public official Google style guide seems to have slightly different style

Honestly I'm suprised to see this b/c most of the repo under Google follows 2-space indentation... 
Another reason why I manually picked 2-space is the on-going unification with TFX, which uses 2-space indentation as well.

**Comment by Ark-kun** [comment]
> Honestly I'm suprised to see this

There are some historical reasons. Some of Google internal code was developed even before PEP-8. That code used 2-space indents and PascalCase function names.
Since most other python projects in the world used snake_case and 4-space indents, Google normally does some massive conversions when releasing an internal product as open-source.
As time goes, more and more projects (ChromeOS, etc) in Google start using the external Google style as opposed to the old internal style for all development.
Since our project is open-source, it makes sense to use the external Google style guide.

**Comment by numerology** [comment]
> > Honestly I'm suprised to see this
> 
> There are some historical reasons. Some of Google internal code was developed even before PEP-8. That code used 2-space indents and `PascalCase` function names.
> Since most other python projects in the world used `snake_case` and 4-space indents, Google normally does some massive conversions when releasing an internal product as open-source.
> As time goes, more and more projects (ChromeOS, etc) in Google start using the external Google style as opposed to the old internal style for all development.
> Since our project is open-source, it makes sense to use the external Google style guide.

Ack. Shall we hold up on this before we get a clearer picture re: the plan of merging? Otherwise we might need to do some back-and-forth conversion, which will be a waste of time.

In the meantime, any other feedbacks will be very helpful :)

**Pull request closed without merging**